### ISSUE TITLE
Can't use type aliases with Option type columns

### ISSUE THREAD

**Issue by beeftornado** [squeryl/squeryl#202]
The following yields a runtime error under scala 2.10.5 and latest release of Squeryl. Replacing `ProductPrice` type with its actual type of `Double` works fine. I like using type aliases because it helps other developers know what the input and outputs are.

```scala
type ProductId = Int
type ProductPrice = Double

case class Product(@Column("productid") id: ProductId,
                   @Column("price") price: Option[ProductPrice]) extends KeyedEntity[ProductId]
```

Error is
```
Caused by: java.lang.RuntimeException: error while reflecting on metadata for (Some(private final scala.Option Product.price),Some(public scala.Option Product.price()),None,Set(@org.squeryl.annotations.ColumnBase(name=, length=-1, optionType=class java.lang.Object, scale=-1, value=price))) of class Product
	at org.squeryl.internals.PosoMetaData$$anonfun$4.apply(PosoMetaData.scala:115)
	at org.squeryl.internals.PosoMetaData$$anonfun$4.apply(PosoMetaData.scala:80)
	at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:224)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:403)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:403)
	at org.squeryl.internals.PosoMetaData.<init>(PosoMetaData.scala:80)
	at org.squeryl.View.<init>(View.scala:66)
	at org.squeryl.Table.<init>(Table.scala:29)
	at org.squeryl.Schema$class.table(Schema.scala:340)
	at storage.Db$.table(Db.scala:12)
	at storage.Db$.<init>(Db.scala:65)
	at storage.Db$.<clinit>(Db.scala)
	... 104 common frames omitted
Caused by: java.lang.RuntimeException: field type scala.None$ is not supported
	at org.squeryl.internals.Utils$.throwError(Utils.scala:95)
	at org.squeryl.internals.FieldMetaData$$anon$2.handleUnknownType(FieldMetaData.scala:643)
	at org.squeryl.internals.FieldMetaData$$anon$2.handleUnknownType(FieldMetaData.scala:602)
	at org.squeryl.internals.FieldTypeHandler$class.handleType(FieldTypeHandler.scala:61)
	at org.squeryl.internals.FieldMetaData$$anon$2.handleType(FieldMetaData.scala:602)
	at org.squeryl.internals.FieldMetaData$.resultSetHandlerFor(FieldMetaData.scala:647)
	at org.squeryl.internals.FieldMetaData.<init>(FieldMetaData.scala:204)
	at org.squeryl.internals.FieldMetaData$$anon$1.build(FieldMetaData.scala:497)
	at org.squeryl.internals.PosoMetaData$$anonfun$4.apply(PosoMetaData.scala:112)
	... 115 common frames omitted
```

**Comment by beeftornado** [comment]
I should note that the Optional type aliasing works if the type resolves to a String. Int, Double, Float, etc. do not work.

**Comment by beeftornado** [comment]
I think I got it to work. Not getting any runtime errors at least. I think the documentation could use some work though as I'm finding lots of undocumented features I have had to explore myself.

Solution was to utilize the `optionType` argument in the `Column` annotation. Initially, I thought `OptionType` was it's own annotation, but that didn't work out well. Then I stumbled upon this test code https://github.com/squeryl/squeryl/blob/master/src/test/scala/org/squeryl/dbagnostic/AnnotationTests.scala that showed other arguments to the `Column` annotation other than the name, which is all I saw discussed in the docs.

Changing the the original code to this has no runtime errors (haven't tried using it yet, but it's a good sign when Squeryl can initialize okay).

```scala
type ProductId = Int
type ProductPrice = Double

case class Product(@Column("productid") id: ProductId,
                   @Column(name = "price", optionType = classOf[ProductPrice]) price: Option[ProductPrice]
) extends KeyedEntity[ProductId]
```
